### PR TITLE
Added microbenchmarks for profiling, correlating atomic performance on QV100

### PR DESCRIPTION
**Pull request by abhaumick**
- atomic_add_bw_conflict
    - added output dump to validate results
- atomic_add_bw_profile
    - some threads conflict on location 0
    - remaining threads access array[gid]
    - kernel arg ConflictCount for testing
- Updated app makefiles

### CODE REVIEW

File: src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw/atomic_add_bw.cu [hunk 1]
```diff
@@ -84,6 +84,8 @@ int main(){
 	gpuErrchk( cudaMemcpy(startClk, startClk_g, TOTAL_THREADS*sizeof(uint32_t), cudaMemcpyDeviceToHost) );
 	gpuErrchk( cudaMemcpy(stopClk, stopClk_g, TOTAL_THREADS*sizeof(uint32_t), cudaMemcpyDeviceToHost) );
 	gpuErrchk( cudaMemcpy(res, res_g, TOTAL_THREADS*sizeof(int32_t), cudaMemcpyDeviceToHost) );
+	printf("Found GPU Data Value = %d %d %d %d\n", data1[0], data1[1], data1[2], data1[3]);
```
Comment (FJShen) [inline src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw/atomic_add_bw.cu:87]: For this and other *.cu files you commited - check if TOTAL_THREADS is greater than 4?

**Comment by abhaumick** [comment]
[7f3fd71](https://github.com/accel-sim/gpu-app-collection/pull/11/commits/7f3fd71cdf51cc8efd4e13932f80938c92b362c5) fixes Makefile issue causing Jenkins fail in [67c2a98](https://github.com/accel-sim/gpu-app-collection/pull/11/commits/67c2a98c69aad94c245c18e6a6578c6339884c13)

File: src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_conflict/atomic_add_bw_conflict_profile.cu [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+	// synchronize all threads
+	asm volatile ("bar.sync 0;");
+
+	// start timing
+	uint32_t start = 0;
+	asm volatile ("mov.u32 %0, %%clock;" : "=r"(start) :: "memory");
+
+	for (int j=0 ; j<REPEAT_TIMES ; ++j) {
+		atomicAdd(&data1[atomic_loc], 10);
```
Comment (FJShen) [inline src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_conflict/atomic_add_bw_conflict_profile.cu]: Where is atomic_loc declared? I don't see it in this *.cu file.

**Review by FJShen — CHANGES_REQUESTED** [review]
Please remove binary file src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_profile/atomic_add_bw_profile away from the repository.

**Comment by abhaumick** [comment]
> Please remove binary file src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_profile/atomic_add_bw_profile away from the repository.

Done.

File: src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_conflict/atomic_add_bw_conflict_profile.cu (hunk 2 above)
Comment (abhaumick) [reply src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_conflict/atomic_add_bw_conflict_profile.cu]: Corrected this ..
Commented out by mistake

**Comment by abhaumick** [comment]
Should not have made the original commit and PR in a hurry.
Mostly cleaned up .. hope no more stupid mistakes remain.

File: src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw/atomic_add_bw.cu (hunk 1 above)
Comment (abhaumick) [reply src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw/atomic_add_bw.cu:87]: Checking the final values is tricky ...
It changes based on the ConflictCount and number of threads active.
I mostly printed it out to make sense in terms of order of magnitude  and by using a calculator (yeah .. sorry)
I can add more prints but unless someone is really digging into the values .. its not straightforward
But .. can be done I feel .. with some effort

**Review by FJShen — APPROVED** [review]

**Review by barnes88 — APPROVED** [review]

**Pull request merged**